### PR TITLE
Make DbgShim really useful for managed attach on Linux

### PR DESCRIPTION
**Pull request by Djuffin**
So far dbgshim didn't do much and exported function in a mangled way. This PR makes it almost as functional as its Windows version.

### CODE REVIEW

**Comment by Djuffin** [comment]
cc @noahfalk , @mikem8361

File: src/pal/prebuilt/inc/mscoree.h [hunk 1]
```diff
@@ -281,7 +281,11 @@ DECLARE_DEPRECATED void STDMETHODCALLTYPE CorExitProcess(int exitCode);
 DEPRECATED_CLR_STDAPI LoadStringRC(UINT iResouceID, _Out_writes_z_(iMax) LPWSTR szBuffer, int iMax, int bQuiet);
 typedef HRESULT (__stdcall *FLockClrVersionCallback) ();
 DEPRECATED_CLR_STDAPI LockClrVersion(FLockClrVersionCallback hostCallback,FLockClrVersionCallback *pBeginHostSetup,FLockClrVersionCallback *pEndHostSetup);
+/*
+On Unix this depricated function produces declaration conflict with another function with the same name from DbgShim.
+We don't expect depricated functions to be used on non-Windows anyway.
 DEPRECATED_CLR_STDAPI CreateDebuggingInterfaceFromVersion(int iDebuggerVersion, LPCWSTR szDebuggeeVersion, IUnknown ** ppCordb);
+*/
```
Comment (jkotas) [inline src/pal/prebuilt/inc/mscoree.h]: This file is auto generated. We do not want to make this manual edit every time the file is refreshed. Could you please find a way how to make this work without manual edits to this auto-generated file?

**Comment by mikem8361** [comment]
Looks good modulo Jan's comments.

**Comment by mikem8361** [comment]
@Djuffin can you make sure "sos ClrStack -i" still works?

**Comment by Djuffin** [comment]
1. sos ClrStack works
2. mscoree.h changes moved to IDL and mscoree.h regenerated
3. debug string format concerns addressed
4. Talked to Noah about function usage

**Pull request merged**